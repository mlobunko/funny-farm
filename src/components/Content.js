import React from "react";
import { connect } from "react-redux";
import Header from "./Header";
import TouchTheAnimal from "./TouchTheAnimal";
import CowCat from "./CowCat";
import Dog from "./Dog";
import DuckSheepChicken from "./DuckSheepChicken";
import { startPlayMusic } from "../actions/saga";

export class Content extends React.Component {
  componentDidMount = () => {
    this.props.startPlayMusic();
  };
  render() {
    return (
      <div>
        {this.props.isLandscape ? (
          <div className="content">
            <Header />
            <TouchTheAnimal />
            <CowCat />
            <Dog />
            <DuckSheepChicken />
          </div>
        ) : null}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLandscape: state.settings.isLandscape
});

export default connect(mapStateToProps, { startPlayMusic })(Content);
