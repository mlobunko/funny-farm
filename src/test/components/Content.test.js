import React from "react";
import { shallow } from "enzyme";
import Content from "../../components/Content";

it("should render Content correctly", () => {
  const wrapper = shallow(<Content />);
  expect(wrapper).toMatchSnapshot();
});
