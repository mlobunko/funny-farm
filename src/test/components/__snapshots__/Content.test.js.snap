// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Content correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component) />,
        <Connect(Component) />,
        <div
          className="animals"
>
          <Connect(Component) />
          <Connect(Component) />
</div>,
      ],
      "className": "content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Component) />,
            <Connect(Component) />,
          ],
          "className": "animals",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component) />,
          <Connect(Component) />,
          <div
            className="animals"
>
            <Connect(Component) />
            <Connect(Component) />
</div>,
        ],
        "className": "content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Component) />,
              <Connect(Component) />,
            ],
            "className": "animals",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
